src/services/SearchService.ts

```diff
@@ -0,0 +1,49 @@
+import type { ISearchItem } from "@/interfaces/ISearchItem";
+import { loadDatesFromLocalStorage, loadFromStorage } from "./StorageService";
+import { formatRegistrations } from "./RegistrationFormatterService";
+
+
+export function searchLocalStorage(searchTerm: string): ISearchItem[] {
+
+    if (searchTerm === "") return [];
+
+    const dates = loadDatesFromLocalStorage()
+
+    const items: ISearchItem[] = [];
+    dates.forEach(date => {
+
+        const lettersToInclude: string[] = [];
+        const dataFromDate: string = loadFromStorage(date)
+
+        const filteredLines = dataFromDate.split(/\r?\n/)
+            .filter((datum) => {
+                const letterMatch = datum.match(/: ([a-zA-Z])/); // Find the letter from a-z or A-Z
+                const letter = letterMatch ? letterMatch[1] : null;
+
+                // Include registration, if search term matches
+                if (datum.includes(searchTerm) && letter) {
```

Comment:
This should be case insensitive

Reply:
Also, maybe you should split the searchterm on spaces, and find all registrations which has all words, so that you can search more "loose"

Reply:
nice catch!

For the search term split, I think we should wait to see if there is a usecase. IMO it could get confusing as to why, you get certain matches

Reply:
Fixed